modules/bullet/btRayShape.cpp

```diff
@@ -97,7 +97,7 @@ void btRayShape::getPreferredPenetrationDirection(int index, btVector3 &penetrat
 
 void btRayShape::reload_cache() {
 
-	m_cacheScaledLength = m_length * m_localScaling[2] + m_collisionMargin;
+	m_cacheScaledLength = m_length * m_localScaling[2];
```

Comment:
Can you please revert this one, we need it I was wrong.

Reply:
Sure, can you explain?

Reply:
Basically in the processCollision function we have this: 

```
if (depth >= -ray_shape->getMargin())
depth = 0;
```
This mean that the margin will be subtracted by the ray size, so if our ray is 1m and the margin is 0.5m the object will penetrate of 0.5 that make the ray half longer.
Instead I'm expecting that the ray stay around 1 and 1.5 so is necessary to tell to the process function that my ray is 1.5 long and allow a penetration of 0.5 (the margin)

Reply:
Yeah, just realized now, testing some of the changes. Sure, if your margin is large this allows convergence. Makes sense.